test/sqllogictest/ldbc_bi.slt

```diff
@@ -0,0 +1,4022 @@
+                            Get l11
+                Project (#0, #1)
+                  Get l10
+            Project (#0, #1, #3)
+              Get l11
+    With
+      cte l11 =
+        Project (#0..=#2, #4)
+          Join on=(#2 = #3) type=differential
+            ArrangeBy keys=[[#2]]
+              Filter (#2) IS NOT NULL
+                Get l10
+            Get l8
+      cte l10 =
+        Union
+          Map (null)
+            Union
+              Negate
+                Project (#0, #1)
+                  Join on=(#0 = #2) type=differential
+                    ArrangeBy keys=[[#0]]
+                      Get l7
+                    ArrangeBy keys=[[#0]]
+                      Distinct group_by=[#0]
+                        Project (#0)
+                          Get l9
+              Get l7
+          Get l9
+      cte l9 =
+        Project (#0, #1, #3)
+          Join on=(#0 = #2) type=differential
+            Get l8
+            ArrangeBy keys=[[#0]]
+              Project (#1, #2)
+                Get materialize.public.person_knows_person
+      cte l8 =
+        ArrangeBy keys=[[#0]]
+          Filter (#0) IS NOT NULL
+            Get l7
+      cte l7 =
+        Project (#3, #4)
+          Map (coalesce(#0, #1), (integer_to_bigint(case when (#0) IS NULL then 0 else 100 end) + coalesce(#2, 0)))
+            Union
+              Project (#2, #0, #1)
+                Map (null)
+                  Union
+                    Negate
+                      Project (#0, #1)
+                        Join on=(#0 = #2) type=differential
+                          Get l4
+                          ArrangeBy keys=[[#0]]
+                            Distinct group_by=[#0]
+                              Get l6
+                    Get l3
+              Map (null, null)
+                Union
+                  Negate
+                    Get l6
+                  Get l2
+              Project (#0, #0, #1)
+                Get l5
+      cte l6 =
+        Project (#0)
+          Get l5
+      cte l5 =
+        Project (#0, #2)
+          Join on=(#0 = #1) type=differential
+            ArrangeBy keys=[[#0]]
+              Get l2
+            Get l4
+      cte l4 =
+        ArrangeBy keys=[[#0]]
+          Get l3
+      cte l3 =
+        Reduce group_by=[#0] aggregates=[count(*)]
+          Project (#15)
+            Filter (#1 = "Abbas_I_of_Persia") AND (#6 < 2010-06-28 00:00:00 UTC) AND (2010-06-14 00:00:00 UTC < #6) AND (#0) IS NOT NULL AND (#15) IS NOT NULL
+              Join on=(#0 = #5 AND #4 = #7 AND #15 = #20) type=differential
+                Get l1
+                ArrangeBy keys=[[#1]]
+                  Project (#1, #2)
+                    Get materialize.public.message_hastag_tag
+                ArrangeBy keys=[[#1]]
+                  Get materialize.public.message
+                Get l0
+      cte l2 =
+        Project (#1)
+          Filter (#14 = "Abbas_I_of_Persia") AND (#1) IS NOT NULL AND (#12) IS NOT NULL
```

Comment:
`Tag.name = 'Abbas_I_of_Persia'` getting stuck again. This seems to be happening way too commonly now.

Reply:
And this one is really weird: The `WHERE Tag.name = 'Abbas_I_of_Persia'` is not on top of any complicated LEFT JOIN or subquery stuff in the original SQL, so I'm really wondering why we can't push it down. We should really investigate this one. It might be that we run `PredicatePushdown` too early when there is still some extra usage of `l1` present that later gets optimized out. In this case, fixing it might be as simple as running `PredicatePushdown` one more time somewhere later in the pipeline.

Reply:
Could you please add some comments to these queries noting these PredicatePushdown anomalies?

Reply:
Added a comment.... but with an index on `Tag.name` it seems like these are getting pushed to a `ReadExistingIndex` appropriately.

Reply:
Ah yes, this one is fine! Without the index on `Tag.name`, it was not pushed so that we can use the index on `Tag.id` for the join.